spec/compiler/codegen/arithmetics_spec.cr

```diff
@@ -1,330 +0,0 @@
-          require "prelude"
-          {{type}}::MAX &+ 1_i64 == {{type}}::MIN
-        )).to_b.should be_true
-      end
-    {% end %}
-  end
-
-  describe "&- subtraction" do
-    {% for type in SupportedInts %}
-      it "wrap around for {{type}}" do
-        run(%(
-          require "prelude"
-          {{type}}::MIN &- {{type}}.new(1) == {{type}}::MAX
-        )).to_b.should be_true
-      end
-
-      it "wrap around for {{type}} - Int64" do
-        run(%(
-          require "prelude"
-          {{type}}::MIN &- 1_i64 == {{type}}::MAX
-        )).to_b.should be_true
-      end
-    {% end %}
-  end
-
-  describe "&* multiplication" do
-    {% for type in SupportedInts %}
-      it "wrap around for {{type}}" do
-        run(%(
-          require "prelude"
-          ({{type}}::MAX // {{type}}.new(2) &+ {{type}}.new(1)) &* {{type}}.new(2) == {{type}}::MIN
-        )).to_b.should be_true
-      end
-
-      it "wrap around for {{type}} + Int64" do
-        run(%(
-          require "prelude"
-          ({{type}}::MAX // {{type}}.new(2) &+ {{type}}.new(1)) &* 2_i64 == {{type}}::MIN
-        )).to_b.should be_true
-      end
-    {% end %}
-  end
-
-  describe "+ addition" do
-    {% for type in SupportedInts %}
-      it "raises overflow for {{type}}" do
-        run(%(
-          require "prelude"
-          begin
-            {{type}}::MAX + {{type}}.new(1)
-            0
-          rescue OverflowError
-            1
-          end
-        )).to_i.should eq(1)
-      end
-
-      it "raises overflow for {{type}} + Int64" do
-        run(%(
-          require "prelude"
-          begin
-            {{type}}::MAX + 1_i64
-            0
-          rescue OverflowError
-            1
-          end
-        )).to_i.should eq(1)
-      end
-    {% end %}
-  end
-
-  describe "- subtraction" do
-    {% for type in SupportedInts %}
-      it "raises overflow for {{type}}" do
-        run(%(
-          require "prelude"
-          begin
-            {{type}}::MIN - {{type}}.new(1)
-            0
-          rescue OverflowError
-            1
-          end
-        )).to_i.should eq(1)
-      end
-
-      it "raises overflow for {{type}} - Int64" do
-        run(%(
-          require "prelude"
-          begin
-            {{type}}::MIN - 1_i64
-            0
-          rescue OverflowError
-            1
-          end
-        )).to_i.should eq(1)
-      end
-    {% end %}
-  end
-
-  describe "* multiplication" do
-    {% for type in SupportedInts %}
-      it "raises overflow for {{type}}" do
-        run(%(
-          require "prelude"
-          begin
-            ({{type}}::MAX // {{type}}.new(2) &+ {{type}}.new(1)) * {{type}}.new(2)
-            0
-          rescue OverflowError
-            1
-          end
-        )).to_i.should eq(1)
-      end
-
-      it "raises overflow for {{type}} * Int64" do
-        run(%(
-          require "prelude"
-          begin
-            ({{type}}::MAX // {{type}}.new(2) &+ {{type}}.new(1)) * 2_i64
```

Comment:
I'm pretty sure this spec is not present from the refactored version. Am I right? And if so, how can we be sure that other specs also haven't disappeared?

Reply:
These are redundant because of:

```crystal
describe "#*" do
  {% for int1 in BUILTIN_INTEGER_TYPES %}
    {% for int2 in BUILTIN_INTEGER_TYPES %}
      run_op_tests {{ int1 }}, {{ int2 }}, :*
    {% end %}
  {% end %}
end
```

Any pair of overflowing values are as good as these. However, I have just added those constants just in case.